src/main/scala/edg_ide/runner/PDFGeneratorUtil.scala

```diff
@@ -1,41 +1,95 @@
 package edg_ide.runner
 
-import com.lowagie.text.{Document, Element, HeaderFooter, Rectangle}
-import com.lowagie.text.pdf.PdfWriter
+import com.lowagie.text.{Anchor, Document, Element, Font, FontFactory, HeaderFooter, Paragraph, Phrase, Rectangle}
+import com.lowagie.text.pdf.{PdfPCell, PdfPTable, PdfWriter}
+import edg.EdgirUtils.SimpleLibraryPath
 import edg.wir.ProtoUtil.BlockProtoToSeqMap
 import edgir.elem.elem.{BlockLike, HierarchyBlock}
 import org.eclipse.elk.graph.ElkNode
 import edg.wir.DesignPath
 import edg_ide.edgir_graph.HierarchyGraphElk.PropertyMapper
 import edg_ide.edgir_graph.{EdgeWrapper, HierarchyGraphElk, NodeDataWrapper, PortWrapper}
 import edg_ide.swing.blocks.ElkNodePainter
+import edgir.ref.ref.LibraryPath
 
 import java.awt.Color
 import java.io.{FileNotFoundException, FileOutputStream, IOException}
-
+import scala.collection.mutable
 
 object PDFGeneratorUtil{
 
+  /*
+    Note for future development:
+      If infinite loop (and eventually StackOverflow) occurs when adding header/footer, check that the
+      page size is large enough/page margin is small enough so that header/footer content do not exceed
+      the available space in the document.
+   */
+  final val PAGE_MARGIN = 15
+  final val font = new Font(Font.HELVETICA, 12f, Font.UNDERLINE, Color.BLUE)
+  final val TABLE_ROW_HEIGHT = 23f
+
+
   private def generatePageSize(node: ElkNode): (Float, Float) = {
-    val width = node.getWidth.toFloat + 2 * ElkNodePainter.margin.toFloat
-    val height = node.getHeight.toFloat + 2 * ElkNodePainter.margin.toFloat
+    val width = node.getWidth.toFloat + 2.5f * ElkNodePainter.margin.toFloat
```

Comment:
Why 2.5? Can this also be a constant like PAGE_MARGIN above?

Reply:
I was tinkering around with the document size before I eventually found out about the error caused by header/footer exceeding into the margin space of the document. I tried to change the margin settings and to set the top coordinates of the header but nothing seemed to fix the issue other than decreasing the margin + increasing the document size... I'll change this to increase the page size by adding the PAGE_MARGIN constant instead though which I think makes more logical sense